app/helpers/search_bar_helper.rb

```diff
@@ -0,0 +1,25 @@
+module SearchBarHelper
+  def mount_search_bar(
+    id:,
+    controller: nil,
+    url: nil,
+    search_query: nil,
+    use_bookmarks: true,
+    use_key_shortcuts: true
+  )
+    bookmarks = {
+      url: api_bookmarks_path,
+      canCreate: authorizer.can?(:create_bookmarks),
+      documentationUrl: documentation_url("4.1.5Searching")
+    }
+    mount_react_component('SearchBar', "##{id}", {
+      controller: controller || auto_complete_controller_name,
```

Comment:
You can set the default value in the kwargs, for example:
```ruby
def mount_search_bar(id:, controller: auto_complete_controller_name, url: "#{auto_complete_controller_name}/auto_complete_search", search_query: params[:search], use_bookmarks: true, use_key_shortcuts: true)

Reply:
Thanks @tbrisker, I tried this before but it didn't work, though it might be the way I consumed this function, looks much better !